### DISCUSSION TITLE
Support for unidirectional protocol / protocol description

### DISCUSSION THREAD

**Discussion by andyboeh** [QuadCorei8085/elero_protocol#5] [Ideas]
There are also unidirectional motors with an older (different) protocol. Have you also had a look at these?

I do have the basic format, but I never figured out how the encryption works. I will check if your decryption can be applied to this as well. 

I also have a lot of SPI traces from a TempoTel 2 that I did back then - old and new transfer format. Some basic protocol description is also available in some FCC documents, are you aware of these?

**Comment by QuadCorei8085** [comment]
I only have access to the system I have installed with my flat. These are 5ch VarioTel2 remotes with venetian blinds / motors.

The FCC documents are a good start to decrypt the protocol segments, however it is not easy to figure out the rolling code. Based on how the Tempotel2 remotes look I would guess they are also using a similar encryption/rolling code but would be better to have a look at the inside, if it is also an atmega with a cc1100/1101 module chances are high it is the same.

**Reply by andyboeh** [reply]
The TempoTel 2 supports both protocols, old and new. We have a mixed system in our house with some motors using the old protocol and some the new one. That's a problem since I controlled the new motors via a Mediola Gateway (Mediola is the OEM of the Centero Gateway; the old version v4 does not require a subscription).

It's a CC1101 connected via SPI to an ATMega (I forgot the exact version), I did quite some traces with a logic analyzer and had a look at it in Sigrok. I think that I got the decoder settings right, but I'm not sure on this. I might have missed the Manchester encoding, now that I had a look at the FCC documents again.

I did all this over three years ago and never figured it out, so thanks to your project, I will give it a fresh start!

**Reply by QuadCorei8085** [reply]
The SPI bus has no Manchester encoding. You wont see that on the logic analyzer. That is a radio thing. Try playing around with the data and see if the code matches somehow. If it is the same HW I doubt the firmware would be so much different.

**Reply by andyboeh** [reply]
I found my old code again: It uses GDO2 as clock and GDO0 as data to transfer Manchester-encoded data. I created a sigrok protocol decoder and I'm pretty sure it works - at least, it correctly detects IDLE1 and IDLE2 as mentioned in the protocol and I get 8 data bytes. 
This is where I stopped the project, other things have taken over since. But now that I found your project, I will revive it!

**Reply by andyboeh** [reply]
@QuadCorei8085 I'm revisiting my project and I was wondering the following: In another discussion, you mentioned that the Atmel wasn't locked on your VarioTel2. Were you able to dump the firmware?
I'm asking because the VarioTel2 also supports the older unidirectional protocol and I still haven't figured out how the obfuscation works.

**Reply by QuadCorei8085** [reply]
Yes the JTAG interface exposed and the MCU is readable. I used the 1wire debug interface because back then i did not notice JTAG although it should have been obvious from the available test pads.

**Reply by andyboeh** [reply]
I sent you an E-Mail to the address you used for commits, maybe it ended up in the spam folder?
